src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -0,0 +1,106 @@
+  private final SparkMax m_motorFollower;
+  private final double kMaxVelocity = 1.75;
+  private final double kMaxAcceleration = 0.75;
+  private final double RotateKMaxVelocity = 1.75;
+  private final double RotateKMaxAcceleration = 0.75;
+  private double m_motorTargetPos = 0, m_motorRotateTarget = 0;
+
+  /** Creates a new intake. */
+  public Intake() {
+    m_motor = new SparkMax(Constants.canIDs.intakeMotor, MotorType.kBrushless);
+    m_motorFollower = new SparkMax(Constants.canIDs.intakeMotorFollower, MotorType.kBrushless);
+    m_motorRotate = new SparkMax(Constants.canIDs.intakeMotorRotate, MotorType.kBrushless);
+
+    SparkMaxConfig config = new SparkMaxConfig();
+    SparkMaxConfig configRotate = new SparkMaxConfig();
+    SparkMaxConfig configFollower = new SparkMaxConfig();
+    config.closedLoop.pidf(
+        Constants.intakeConstants.kP,
+        Constants.intakeConstants.kI,
+        Constants.intakeConstants.kD,
+        Constants.intakeConstants.kFF);
+
+    config
+        .closedLoop
+        .maxMotion
+        .maxAcceleration(kMaxAcceleration)
+        .maxVelocity(kMaxVelocity)
+        .positionMode(MAXMotionPositionMode.kMAXMotionTrapezoidal);
+
+    config.encoder.positionConversionFactor(2.0 * Math.PI * 1.5);
+
+    m_motor.configure(config, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+
+    configRotate.closedLoop.pidf(
+        Constants.intakeConstants.RotateKP,
+        Constants.intakeConstants.RotateKI,
+        Constants.intakeConstants.RotateKD,
+        Constants.intakeConstants.RotateKFF);
+
+    configRotate
+        .closedLoop
+        .maxMotion
+        .maxAcceleration(RotateKMaxAcceleration)
+        .maxVelocity(RotateKMaxVelocity)
+        .positionMode(MAXMotionPositionMode.kMAXMotionTrapezoidal);
+
+    configRotate.encoder.positionConversionFactor(2.0 * Math.PI * 1.5);
+
+    m_motorRotate.configure(
+        configRotate, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+
+    configFollower.follow(m_motor, true);
+
+    m_motorFollower.configure(
+        configFollower, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+  }
+
+  private void goToGoal(double goal) {
+    m_motorTargetPos = goal;
+    m_motorRotate
+        .getClosedLoopController()
+        .setReference(goal, ControlType.kMAXMotionPositionControl);
+  }
+
+  private void upSpool() {
+    m_motorRotateTarget = 1;
+    m_motor.getClosedLoopController().setReference(1, ControlType.kMAXMotionVelocityControl);
+  }
+
+  private void downSpool() {
+    m_motorRotateTarget = -1;
+    m_motor.getClosedLoopController().setReference(-1, ControlType.kMAXMotionVelocityControl);
+  }
+
+  @Override
+  public void periodic() {
+    Logger.recordOutput("M_Motor Error", m_motorTargetPos - m_motor.getEncoder().getPosition());
+    Logger.recordOutput(
+        "M_MotorRotate Error", m_motorRotateTarget - m_motorRotate.getEncoder().getPosition());
+  }
+
+  public Command spoolUp() {
```

Comment:
make sure we have commands to:
- `intake`
- `spitOut`
- `raiseIntake`
- `lowerIntake`